bin/update-ips.bash

```diff
@@ -184,6 +212,69 @@ yq_dig() {
   echo "$3"
 }
 
+# yq_dig_secrets <node> <default>
+yq_dig_secrets() {
+    ret=$(sops -d "${secrets["secrets_file"]}" | yq4 "$1")
+
+    if [[ "$ret" != "null" ]]; then
+        echo "$ret"
+        return
+    fi
+
+  echo "$2"
+}
+
+get_swift_url() {
+    local auth_url
+    auth_url="$(yq_dig 'sc' '.objectStorage.swift.authUrl' '""')"
+
+    if [ -n "$(yq_dig_secrets '.objectStorage.swift.username' "")" ]; then
+        response=$(curl -i -s -H "Content-Type: application/json" -d '
+        {
+          "auth": {
+            "identity": {
+              "methods": ["password"],
+              "password": {
+                "user": {
+                  "name": "'"$(yq_dig_secrets '.objectStorage.swift.username' '""')"'",
+                  "domain": { "id": "'"$(yq_dig '.objectStorage.swift.domainName' '""')"'" },
+                  "password": "'"$(yq_dig_secrets '.objectStorage.swift.password' '""')"'"
+                }
+              }
+            },
+            "scope": {
+              "project": {
+                "name": "'"$(yq_dig '.objectStorage.swift.projectName' '""')"'",
+                "domain": { "name": "'"$(yq_dig '.objectStorage.swift.projectDomainName' '""')"'" }
+              }
+            }
+          }
+        }' "${auth_url}/auth/tokens")
+    elif [ -n "$(yq_dig_secrets '.objectStorage.swift.applicationCredentialID' "")" ]; then
+        response=$(curl -i -s -H "Content-Type: application/json" -d '
+        {
+          "auth": {
+            "identity": {
+              "methods": ["application_credential"],
+              "application_credential": {
+                "id": "'"$(yq_dig_secrets '.objectStorage.swift.applicationCredentialID' '""')"'",
+                "secret": "'"$(yq_dig_secrets '.objectStorage.swift.applicationCredentialSecret' '""')"'"
+              }
+            }
+          }
+        }' "${auth_url}/auth/tokens")
+    fi
+
+    local os_token
+    local swift_url
+    os_token=$(echo "$response" | grep -oP "x-subject-token:\s+\K\S+")
+    swift_url=$(echo "$response" | tail -n +15 | jq -r '.[].catalog[] | select( .type == "object-store" and .name == "swift") | .endpoints[] | select(.interface == "public") | .url')
+
+    curl -i -s -X DELETE -H "X-Auth-Token: $os_token" -H "X-Subject-Token: $os_token" "${auth_url}/auth/tokens" > /dev/null
```

Comment:
Why do you need to delete this token?

Reply:
I think it's good not to have tokens laying around.
I think they are valid for 12 or 24 hours.